app/src/main/java/de/westnordost/streetcomplete/quests/smoking/AddSmoking.kt

```diff
@@ -19,7 +19,7 @@ class AddSmoking : OsmFilterQuestType<SmokingAllowed>() {
         return """
               ${p}amenity ~ bar|cafe|pub|biergarten|restaurant|food_court|nightclub|stripclub
               or ${p}leisure ~ outdoor_seating
-              or (${p}amenity ~ fast_food|ice_cream or ${p}shop ~ ice_cream|deli|bakery|coffee|tea|wine)
+              or ((${p}amenity ~ fast_food|ice_cream) or (${p}shop ~ ice_cream|deli|bakery|coffee|tea|wine))
```

Comment:
this one is a bit too much, `key~value|list` is inherently one element

Reply:
yes, I might have overdone it to be absolutely sure for debug... but **_still it does not work_**:

`.apk` from this build  https://github.com/mnalis/StreetComplete/actions/runs/1907292270 which corresponds to this file https://github.com/mnalis/StreetComplete/blob/e2e103dbbbd774ea40a5642a95a81a5b466a95eb/app/src/main/java/de/westnordost/streetcomplete/quests/smoking/AddSmoking.kt#L17-L39

It still activates a quest on this node, and it shouldn't:  https://www.openstreetmap.org/node/4993598032

Reply:
@mnalis 

Are you sure that you run latest code and it is pushed? I fetched it to debug what is going on and it failed on compilation error, with `IS_SHOP_OR_DISUSED_SHOP_EXPRESSION` not being defined.

https://github.com/streetcomplete/StreetComplete/pull/3784/files appears to be not defined in your branch.

It is defined in latest StreetComplete code, but it is not yet merged in to your branch.

After `git merge master` compilation succeeds.

Reply:
@mnalis 

Are you sure that you run latest code and it is pushed? I fetched it to debug what is going on and it failed on compilation error, with `IS_SHOP_OR_DISUSED_SHOP_EXPRESSION` not being defined.

https://github.com/streetcomplete/StreetComplete/pull/3784/files appears to be not defined in your branch.

It is defined in latest StreetComplete code, but it is not yet merged in to your branch.

After `git merge master` compilation succeeds.

I build code from such version and https://www.openstreetmap.org/node/9200004349 was ineligible.

Reply:
@mnalis 

Are you sure that you run latest code and it is pushed? I fetched it to debug what is going on and it failed on compilation error, with `IS_SHOP_OR_DISUSED_SHOP_EXPRESSION` not being defined.

https://github.com/streetcomplete/StreetComplete/pull/3784/files appears to be not defined in your branch.

It is defined in latest StreetComplete code, but it is not yet merged in to your branch.

After `git merge master` compilation succeeds.

I build code from such version and https://www.openstreetmap.org/node/9200004349 was ineligible.

> It still activates a quest on this node, and it shouldn't: https://www.openstreetmap.org/node/4993598032

It does not for me. It activated for nearby https://www.openstreetmap.org/node/1917302043

Reply:
It is also not activated for https://www.openstreetmap.org/node/4993597838

(nevertheless, I am not sure about requiring seating to be tagged, but that is a separate topic)